### PR TITLE
[NTFS] New defines & stub for reading encrypted files

### PR DESCRIPTION
**Pull request by nagayev**
## Purpose
We shoudn't read file if it's encrypted.
Also I added several new defines for future contributions.

## Proposed changes
- Add new defines for NTFS's file attributes
- Don't read file if it's encrypted

### CODE REVIEW

File: drivers/filesystems/ntfs/fcb.c [hunk 1]
```diff
@@ -143,6 +143,12 @@ NtfsFCBIsCompressed(PNTFS_FCB Fcb)
     return ((Fcb->Entry.FileAttributes & NTFS_FILE_TYPE_COMPRESSED) == NTFS_FILE_TYPE_COMPRESSED);
 }
 
+BOOLEAN
+NtfsFCBIsEncrypted(PNTFS_FCB Fcb)
+{
+    return ((Fcb->Entry.FileAttributes & NTFS_FILE_TYPE_ENCRYPTED) == NTFS_FILE_TYPE_ENCRYPTED);
```
Comment (SergeGautherie) [inline drivers/filesystems/ntfs/fcb.c:149]: The `==` part seems overkill to check one bit only, even though some existing code does it that way.

File: drivers/filesystems/ntfs/ntfs.h [hunk 2]
```diff
@@ -931,6 +935,9 @@ NtfsFCBIsReparsePoint(PNTFS_FCB Fcb);
 BOOLEAN
 NtfsFCBIsCompressed(PNTFS_FCB Fcb);
 
+BOOLEAN
+NtfsFCBIsEncrypted(PNTFS_FCB Fcb);
+
```
Comment (SergeGautherie) [inline drivers/filesystems/ntfs/ntfs.h:940]: Move this part to the second commit.

Comment (nagayev) [reply drivers/filesystems/ntfs/ntfs.h:940]: okey

File: drivers/filesystems/ntfs/fcb.c (hunk 1 above)
Comment (ColinFinck) [reply drivers/filesystems/ntfs/fcb.c:149]: It's consistent with the existing code, so that should be fine

**Review by ColinFinck — APPROVED** [review]

**Comment by Extravert-ir** [comment]
Don't quite get why it was needed to create a separate function for checking every individual flag, but that's not a topic of this PR

**Pull request merged**